src/include/homestore/replication/repl_dev.h

```diff
@@ -61,51 +69,89 @@ struct repl_req_ctx : public boost::intrusive_ref_counter< repl_req_ctx, boost::
     friend class SoloReplDev;
 
 public:
-    repl_req_ctx() { start_time = Clock::now(); }
+    repl_req_ctx() { m_start_time = Clock::now(); }
     virtual ~repl_req_ctx();
-    int64_t get_lsn() const { return lsn; }
-    MultiBlkId const& get_local_blkid() const { return local_blkid; }
+    void init(repl_key rkey, journal_type_t op_code, bool is_proposer, sisl::blob const& user_header,
+              sisl::blob const& key, uint32_t data_size);
+
+    /////////////////////// All getters ///////////////////////
+    repl_key const& rkey() const { return m_rkey; }
+    uint64_t dsn() const { return m_rkey.dsn; }
+    uint64_t term() const { return m_rkey.term; }
+    int64_t lsn() const { return m_lsn; }
+    bool is_proposer() const { return m_is_proposer; }
+    journal_type_t op_code() const { return m_op_code; }
+
+    sisl::blob const& header() const { return m_header; }
+    sisl::blob const& key() const { return m_key; }
+    MultiBlkId const& local_blkid() const { return m_local_blkid; }
+    RemoteBlkId const& remote_blkid() const { return m_remote_blkid; }
+    const char* data() const { return r_cast< const char* >(m_data); }
+    repl_req_state_t state() const { return repl_req_state_t(m_state.load()); }
+    bool has_state(repl_req_state_t s) const { return m_state.load() & uint32_cast(s); }
+    repl_journal_entry const* journal_entry() const { return m_journal_entry; }
+    uint32_t journal_entry_size() const;
+    bool is_localize_pending() const { return m_is_jentry_localize_pending; }
+    bool is_data_inlined() const { return (m_op_code == journal_type_t::HS_DATA_INLINED); }
+    bool has_linked_data() const { return (m_op_code == journal_type_t::HS_DATA_LINKED); }
 
-    uint64_t dsn() const { return rkey.dsn; }
-    uint64_t term() const { return rkey.term; }
-    void alloc_journal_entry(uint32_t size, bool is_raft_buf);
     raft_buf_ptr_t& raft_journal_buf();
     uint8_t* raw_journal_buf();
+    flatbuffers::FlatBufferBuilder& create_fb_builder() { return m_fb_builder; }
+    void release_fb_builder() { m_fb_builder.Release(); }
 
+    /////////////////////// Non modifiers methods //////////////////
     std::string to_string() const;
     std::string to_compact_string() const;
-    Clock::time_point created_time() const { return start_time; }
+    Clock::time_point created_time() const { return m_start_time; }
+
+    /////////////////////// All Modifiers methods //////////////////
+    ReplServiceError alloc_local_blks(cshared< ReplDevListener >& listener, uint32_t data_size);
```

Comment:
Can we put api documents here for some modifiers for readability for whom that are not familiar with this code?

Reply:
sure will do.